core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -804,7 +809,6 @@ class KafkaController(val config: KafkaConfig,
     replicaStateMachine.handleStateChanges(replicasToBeDeleted.toSeq, OfflineReplica)
     // send stop replica command to the old replicas
     replicaStateMachine.handleStateChanges(replicasToBeDeleted.toSeq, ReplicaDeletionStarted)
-    // TODO: Eventually partition reassignment could use a callback that does retries if deletion failed
```

Comment:
Opened https://issues.apache.org/jira/browse/KAFKA-8632 so we don't have it in a TODO.